install_scripts/linux-python3/tools.sh

```diff
@@ -0,0 +1,42 @@
+# Subroutines to call esptool with common arguments
+
+# Ensure local binary path is in $PATH (pip3 default target path)
+echo "$PATH" | grep '$HOME\/\.local\/bin' 2>&1 >/dev/null
+if test "$?" != "0"; then
+    export PATH="$HOME/.local/bin:$PATH"
+fi
+
+# Install esptool.py if needed
+which esptool.py 2>&1 >/dev/null
+if test "$?" != "0"; then
+    echo esptool.py not found, attempting to install
+    python3 -m pip install --user esptool
```

Comment:
Instead of installing esptool with pip, how about we just include esptool.py in the bundle, to ensure that we get the version that we want.

Reply:
That is a *very* good idea, especialy given ESP's penchant for putting breaking changes into minor revision updates..

I'll look at this properly tomorrow, the install-python3 script can grab the source zip in the same way as the other individual platform zips, and then `pip install <downloaded src>.zip` (I just tested this, and pip installing a zipfile directly works fine on the commandline)

Reply:
I just pushed #[b12c76c](https://github.com/bdring/FluidNC/pull/272/commits/b12c76cc6ddf13287bff7da6674c7a271d540449) to the request; this copies the correct source bundle to the package and tools.sh then install from that, rather than pipy.